pkg/server/cloud/common.go

```diff
@@ -140,3 +140,10 @@ func ToSaneVolumeSize(volSizeSpec string) int32 {
 	}
 	return volSizeGiB
 }
+
+func AddVolSpecSizeToRootSize(specSize int32, img Image) int32 {
+	if img.VolumeDiskSize == nil {
+		return specSize
+	}
+	return *img.VolumeDiskSize + specSize
```

Comment:
Peeking in on your work in progress: I think what we want to do here is make sure we don't request a disk size (the `specSize`) that's less than the actual volume size for the disk since that causes boot errors. No need to add, just return the larger of `img.VolumeDiskSize` and `specSize`